### PR TITLE
bug fixes

### PR DESCRIPTION
**Pull request by yarneo**
1. in decide response if errored makes semaphore stuck. 
2. crash in archive if an object doesnt implement NSCoder. 
3. archiveProperties after doing a superProperty action to save from possible loss of data.

### CODE REVIEW

File: Mixpanel/Mixpanel.m [hunk 1]
```diff
@@ -347,9 +347,7 @@ - (void)registerSuperProperties:(NSDictionary *)properties
         NSMutableDictionary *tmp = [NSMutableDictionary dictionaryWithDictionary:self.superProperties];
         [tmp addEntriesFromDictionary:properties];
         self.superProperties = [NSDictionary dictionaryWithDictionary:tmp];
-        if ([Mixpanel inBackground]) {
```
Comment (yinfeiru) [inline Mixpanel/Mixpanel.m]: why we remove this check?

File: Mixpanel/Mixpanel.m [hunk 2]
```diff
@@ -1122,6 +1126,7 @@ - (void)checkForDecideResponseWithCompletion:(void (^)(NSArray *surveys, NSArray
                     if (completion) {
                         completion(nil, nil, nil, nil);
                     }
+                    dispatch_semaphore_signal(semaphore);
```
Comment (bclymer) [inline Mixpanel/Mixpanel.m]: I think these `dispatch_semaphore_signal` should be in a separate PR so it could be merged with less risk. This definitely seems like a bug, as the semaphore lock will never go away if the device is offline during init.

File: Mixpanel/Mixpanel.m (hunk 1 above)
Comment (yarneo) [inline Mixpanel/Mixpanel.m]: Yep!
